mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/ui/LinkBankAccountScreen.kt

```diff
@@ -0,0 +1,327 @@
+
+                    Text(
+                        text = "Popular Banks",
+                        style = TextStyle(Color.Black, fontWeight = FontWeight.Medium),
+                        modifier = Modifier.padding(start = 10.dp)
+                    )
+                    Spacer(modifier = Modifier.height(20.dp))
+                    BankGrid(banks = popularBanks, onSelectBank)
+                }
+
+
+
+            if (filteredBanks.value.isNotEmpty())
+                Column {
+                    if (search.value.isBlank()) {
+                        Spacer(modifier = Modifier.height(20.dp))
+                        Text(
+                            text = "Other Banks",
+                            style = TextStyle(Color.Black, fontWeight = FontWeight.Medium),
+                            modifier = Modifier.padding(start = 10.dp)
+                        )
+                        Spacer(modifier = Modifier.height(20.dp))
+                    }
+
+                    BankList(
+                        otherBanks = filteredBanks.value,
+                        onSelectBank
+                    )
+                }
+            else
+                Box(
+                    modifier = Modifier.fillMaxWidth().padding(top = 15.dp),
+                    contentAlignment = Alignment.Center
+                ) {
+                    Text(
+                        text = "No bank found",
+                        style = MaterialTheme.typography.subtitle1
+                    )
+                }
+        }
+    }
+}
+
+
+@Composable
+fun BankGrid(banks: List<Bank>, onSelectBank: (Bank) -> Unit) {
+    Box(
+        modifier = Modifier
+            .fillMaxWidth()
+            .height(200.dp)
+            .background(Color.White)
+            .shadow(elevation = 2.dp)
+    ) {
+        LazyVerticalGrid(columns = GridCells.Fixed(3)) {
+            items(banks.size) { index ->
+                BankItem(bank = banks[index], onSelectBank)
+            }
+        }
+    }
+}
+
+@Composable
+fun BankItem(bank: Bank, onSelectBank: (Bank) -> Unit) {
+    Column(
+        horizontalAlignment = Alignment.CenterHorizontally,
+        modifier = Modifier
+            .padding(8.dp)
+            .height(80.dp)
+            .clickable {
+                onSelectBank(bank)
+            }
+    ) {
+        Image(
+            painter = painterResource(id = bank.image),
+            contentDescription = bank.name,
+            modifier = Modifier.weight(1f)
+        )
+        Text(
+            text = bank.name,
+            style = MaterialTheme.typography.body1.copy(fontSize = 12.sp),
+            modifier = Modifier
+
+                .padding(top = 4.dp)
+        )
+    }
+}
+
+@Composable
+fun BankList(otherBanks: List<Bank>, onSelectBank: (Bank) -> Unit) {
+
+    Box(
+        modifier = Modifier
+            .background(Color.White)
+    ) {
+        Column(
+        ) {
+            otherBanks.forEachIndexed { index, bank ->
```

Comment:
use lazy column here as well

Reply:
@PratyushSingh07 I used LazyColumn here but i got a unbounded constraint error since it is nested inside a column. and the scroll behavior will not replicate the original one (Like the XML). We can add height to the lazycolumn but that will not be optimal.

Reply:
let me know if we can use flow columns here